core/src/main/scala/kafka/coordinator/transaction/TransactionMarkerRequestCompletionHandler.scala

```diff
@@ -155,6 +155,14 @@ class TransactionMarkerRequestCompletionHandler(brokerId: Int,
                       txnMarkerChannelManager.removeMarkersForTxnId(transactionalId)
                       abortSending = true
 
+                    case Errors.UNSUPPORTED_FOR_MESSAGE_FORMAT =>
```

Comment:
Can we handle UNSUPPORTED_VERSION here as well? Considering your other patch which checks the inter broker protocol version, it is possible (if unlikely) that we could hit that error in the middle of a rolling upgrade.